addons/mail/static/src/models/message_action_list.js

```diff
@@ -1,187 +1,163 @@
+import { attr, many, one } from '@mail/model/model_field';
 import { clear, insertAndReplace } from '@mail/model/model_field_command';
-import { markEventHandled } from '@mail/utils/utils';
 
 registerModel({
     name: 'MessageActionList',
     identifyingFields: ['messageView'],
     recordMethods: {
         /**
          * @private
-         * @param {MouseEvent} ev
+         * @returns {integer}
          */
-        onClick(ev) {
-            markEventHandled(ev, 'MessageActionList.Click');
+        _computeActionViewsWithoutCompactCount() {
+            return this.messageActionViews.filter((action) => !action.messageActionListOwnerAsToggleCompact).length;
         },
         /**
          * @private
-         * @param {MouseEvent} ev
+         * @returns {FieldCommand}
          */
-        onClickDelete(ev) {
-            this.update({ deleteConfirmDialog: insertAndReplace() });
-        },
-        /**
-         * @private
-         * @param {MouseEvent} ev
-         */
-         onClickEdit(ev) {
-            this.messageView.startEditing();
-        },
-        /**
-         * @private
-         * @param {MouseEvent} ev
-         */
-        onClickMarkAsRead(ev) {
-            this.message.markAsRead();
-        },
-        /**
-         * Handles click on the reaction icon.
-         */
-        onClickActionReaction() {
-            if (!this.reactionPopoverView) {
-                this.update({ reactionPopoverView: insertAndReplace() });
-            } else {
-                this.update({ reactionPopoverView: clear() });
+        _computeDeleteActionView() {
+            if (this.message && this.message.canBeDeleted) {
+                return insertAndReplace();
             }
-        },
-        /**
-         * @param {MouseEvent} ev
-         */
-        onClickReaction(ev) {
-            this.message.addReaction(ev.currentTarget.dataset.unicode);
-            this.update({ reactionPopoverView: clear() });
-        },
-        /**
-         * Opens the reply composer for this message (or closes it if it was
-         * already opened).
-         *
-         * @private
-         * @param {MouseEvent} ev
-         */
-        onClickReplyTo(ev) {
-            markEventHandled(ev, 'MessageActionList.replyTo');
-            this.messageView.replyTo();
+            return clear();
         },
         /**
          * @private
-         * @param {MouseEvent} ev
+         * @returns {FieldCommand}
          */
-        onClickToggleCompact(ev) {
-            this.update({ isCompact: !this.isCompact });
+        _computeEditActionView() {
+            if (this.message && this.message.canBeDeleted) {
+                return insertAndReplace();
+            }
+            return clear();
         },
         /**
          * @private
-         * @param {MouseEvent} ev
+         * @returns {boolean}
          */
-        onClickToggleStar(ev) {
-            this.message.toggleStar();
+        _computeHasReplyToAction() {
```

Comment:
I think this could directly be the compute for the action view, no need to store an extra boolean?